### PR TITLE
Only use WKKeyedCoder for encoding when secure coding class is not specified

### PR DESCRIPTION
**Pull request by szewai**
#### 8b4884931b2c0f8412cff37d9b8ae7724736b2f2
<pre>
Only use WKKeyedCoder for encoding when secure coding class is not specified
<a href="https://bugs.webkit.org/show_bug.cgi?id=279457">https://bugs.webkit.org/show_bug.cgi?id=279457</a>
<a href="https://rdar.apple.com/135631932">rdar://135631932</a>

Reviewed by Alex Christensen.

The existing implementation expects _webKitPropertyListData to return a dictionary that matches the declaration in
existing CoreIPC*.serialization.in file, when _webKitPropertyListData selector exists. However, a class could add
support for _webKitPropertyListData independently, and the result may not match the existing declaration, which could
lead to breakage in existing WebKit clients. To avoid such issue, now we only allow to use WKKeyedCoder for types that
have WKKeyedCoder support and do not specify secure coding class. When a class adds implementation for
_webKitPropertyListData, the WKKeyedCoder coding will still be in use, and we could explicitly enable the new encoding
format by updating CoreIPC*.serialization.in file and specifying secure encoding class later.

Also, update tests with new auto-generated files.

* Source/WebKit/Scripts/generate-serializers.py:
(generate_webkit_secure_coding_impl):
(generate_webkit_secure_coding_impl.is):
* Source/WebKit/Scripts/webkit/messages_unittest.py:
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/GeneratedWebKitSecureCoding.cpp:
(WebKit::dictionaryForWebKitSecureCodingTypeFromWKKeyedCoder):
(WebKit::dictionaryForWebKitSecureCodingType):
(WebKit::CoreIPCAVOutputContext::toID const):
(WebKit::CoreIPCNSSomeFoundationType::toID const):
(WebKit::NSSomeOtherFoundationType):
(WebKit::NSSomeOtherFoundationType::toID const):
(WebKit::CoreIPCDDScannerResult::toID const):
* Source/WebKit/Scripts/webkit/tests/GeneratedWebKitSecureCoding.h:
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::jsValueForArguments):
(IPC::messageArgumentDescriptions):
* Source/WebKit/Scripts/webkit/tests/MessageNames.cpp:
* Source/WebKit/Scripts/webkit/tests/MessageNames.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Scripts/webkit/tests/WebKitPlatformGeneratedSerializers.cpp:
(IPC::NSSomeOtherFoundationType&gt;::encode):
(IPC::NSSomeOtherFoundationType&gt;::decode):

Canonical link: <a href="https://commits.webkit.org/283566@main">https://commits.webkit.org/283566@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/a1e641584fc6f2b35ba74aa908d39ed900cce3d2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66537 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45912 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19158 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70570 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/17670 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53711 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17430 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53325 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11918 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69604 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42275 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57576 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33986 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/66049 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38946 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16024 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60855 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15306 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72273 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10494 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14677 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60654 "Found 1 new test failure: webanimations/accelerated-transform-animation-from-scale-zero-and-implicit-to-kefyrame.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10526 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57645 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60976 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14710 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8636 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2254 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41719 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42796 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43979 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42539 "Built successfully") | | |

### CODE REVIEW

**Comment by sheeparegreat** [comment]
Typo: Avoing -> Avoid

File: Source/WebKit/Scripts/webkit/messages_unittest.py [hunk 1]
```diff
@@ -55,6 +55,8 @@
     'TestWithStreamServerConnectionHandle',
     'TestWithEnabledBy',
     'TestWithEnabledIf',
+    'TestWithEnabledByAndConjunction',
```
Comment (sheeparegreat) [inline Source/WebKit/Scripts/webkit/messages_unittest.py:58]: Are the two changes here related to the generate-serializers.py change? These don't seem to be related to the maySupportWebKitSecureEncoding part?

**Comment by szewai** [comment]
> Typo: Avoing -> Avoid

Fixed!

File: Source/WebKit/Scripts/webkit/messages_unittest.py (hunk 1 above)
Comment (szewai) [reply Source/WebKit/Scripts/webkit/messages_unittest.py:58]: The patch updates the files generated by `make -C Source/WebKit/Scripts/webkit/tests`, as it has an effect on the generated content (e.g. new function is named `dictionaryForWebKitSecureCodingTypeFromWKKeyedCoder`). The auto generated files include message receivers for `TestWithEnabledByAndConjunction` and `TestWithEnabledByOrConjunction`, as there are corresponding messages.in files.
After this change, a unit test that checks the existing message receivers fails, because these two message receivers are not listed in expectation, so this change is updating the expectation so that the test can pass.
Technically the unit test should be fixed when the messages.in files were added; and it's now fixed by my patch.

**Review by achristensen07 — APPROVED** [review]

**Review by sheeparegreat — APPROVED** [review]
